macfuse: add fs_link variant

i0ntempest opened the pull request:
#### Description

<!-- Note: it is best to make pull requests from a branch rather than from master -->

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion) $(uname -m)"; xcode=$(xcodebuild -version 2>/dev/null); if [ $? == 0 ]; then echo "$(echo "$xcode" | awk '\''NR==1{x=$0}END{print x" "$NF}'\'')"; else echo "Command Line Tools $(pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | awk '\''/version:/ {print $2}'\'')"; fi' | tee /dev/tty | pbcopy
-->
macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

i0ntempest commented:
I can also put this into a variant if we don't want to violate mtree by default.

mohd-akram commented:
A user might have installed macFUSE from their website. We don't want to mess with that. It's this way on purpose.

mohd-akram commented:
> I can also put this into a variant if we don't want to violate mtree by default.

This step needs to only be done once and never again. I don't think we need a variant for it.

i0ntempest commented:
> > I can also put this into a variant if we don't want to violate mtree by default.
> 
> This step needs to only be done once and never again. I don't think we need a variant for it.

All files being handled my MacPorts would eliminate the possibility of a broken symlink if the user removes the port. Also better user experience.
Just my 2 cent tho, I'm not going to force it if you think it's really unnecessary.

i0ntempest commented:
@mohd-akram I added a few clean up steps in destroot to remove more unneeded files. Are you okay with the changes here?
Users will not need to do anything unless they choose to opt in this variant. Also I've added the same fs_link variant to a few other fuse filesystems.

mohd-akram commented on fuse/macfuse/Portfile:
```diff
@@ -61,23 +61,40 @@ build {
     xinstall -W ${pkg}${dir}/Contents/Resources -m 4755 load_macfuse mount_macfuse \
         ${destroot}${prefix}${dir}/Contents/Resources
+    delete {*}[glob ${destroot}${prefix}${dir}/Contents/Resources/*.app]
+    delete ${destroot}${prefix}${dir}/Contents/_CodeSignature
```
I believe I didn't delete these files to not invalidate the signature. IMO remove this for now and just keep the `fs_link` change in this PR.

i0ntempest replied:
I believe only the signature of kexts is validated, deleting the signature in the fs bundle shouldn't matter. I'll spin up a vm and test real quick

mohd-akram replied:
TBH I'd rather not mess with signatures, especially since this is outside the MacPorts root.

mohd-akram commented on fuse/macfuse/Portfile:
```diff
@@ -61,23 +61,40 @@ build {
         sudo ln -fsn ${prefix}${dir} ${dir}
-"
+    
+    Or alternatively use +fs_link variant to automatically create this symlink.
```
```suggestion
    Alternatively, use the +fs_link variant to automatically create this symlink.
```

mohd-akram replied:
Not quite resolved.

i0ntempest replied:
Should be ok now

mohd-akram commented on fuse/macfuse/Portfile:
```diff
@@ -73,11 +74,25 @@ destroot {
 
         sudo ln -fsn ${prefix}${dir} ${dir}
-"
+    
```
Could you remove this whitespace?

i0ntempest replied:
Done

mohd-akram reviewed (approved):
Thanks!
